packages/manager/apps/pci-public-ip/src/pages/order/steps/FloatingSteps.tsx

```diff
@@ -180,92 +177,106 @@ export const FloatingSteps = ({
         {...steps.get(StepIdsEnum.FLOATING_INSTANCE)}
         title={tOrder('pci_additional_ip_create_step_attach_instance')}
         next={{ action: form.instance && form.ipAddress && On.next }}
+        onEdit={On.edit}
         order={3}
       >
-        {selectedRegionInstances.length !== 0 ? (
+        {isInstanceFetching ? (
+          <div className="text-center">
+            <OsdsSpinner inline />
+          </div>
+        ) : (
           <>
-            <p>
-              <OsdsText>
-                {tOrder(
-                  'pci_additional_ip_create_step_attach_instance_description_floating_ip',
-                )}
-              </OsdsText>
-            </p>
-            <div>
-              <OsdsText>
-                {tOrder('pci_additional_ips_failoverip_order_instance')}
-              </OsdsText>
-            </div>
-            <OsdsSelect
-              className="mb-4"
-              required
-              value={form.instance?.id}
-              onOdsValueChange={(event) => {
-                setForm({
-                  ...form,
-                  instance: getInstanceById(event.detail.value as string),
-                });
-              }}
-            >
-              <span slot="placeholder">
-                {tOrder('pci_additional_ip_create_step_attach_instance_label')}
-              </span>
-              {selectedRegionInstances.map((instance) => (
-                <OsdsSelectOption key={instance.id} value={instance.id}>
-                  {instance.name}
-                </OsdsSelectOption>
-              ))}
-            </OsdsSelect>
-            {form.instance && (
+            {selectedRegionInstances.length !== 0 ? (
               <>
+                <p>
+                  <OsdsText>
+                    {tOrder(
+                      'pci_additional_ip_create_step_attach_instance_description_floating_ip',
+                    )}
+                  </OsdsText>
+                </p>
                 <div>
                   <OsdsText>
-                    {tOrder('pci_additional_ip_select_network_label')}
+                    {tOrder('pci_additional_ips_failoverip_order_instance')}
                   </OsdsText>
                 </div>
                 <OsdsSelect
+                  className="mb-4"
                   required
-                  value={form.ipAddress?.ip}
+                  value={form.instance?.id}
                   onOdsValueChange={(event) => {
                     setForm({
                       ...form,
-                      ipAddress: form.instance.ipAddresses.find(
-                        (ipAddress) =>
-                          ipAddress.ip === (event.detail.value as string),
-                      ),
+                      instance: getInstanceById(event.detail.value as string),
```

Comment:
Better to stringify `value` instead of this type cast:
```suggestion
                     instance: getInstanceById(String(event.detail.value)),
```